public/app/core/components/RolePicker/UserRolePicker.tsx

```diff
@@ -26,29 +26,34 @@ export const UserRolePicker: FC<Props> = ({
   disabled,
   builtinRolesDisabled,
 }) => {
-  const [appliedRoles, setAppliedRoles] = useState<Role[]>([]);
-
-  const { loading } = useAsync(async () => {
+  const [{ loading, value: appliedRoles }, getUserRoles] = useAsyncFn(async () => {
     try {
       if (contextSrv.hasPermission(AccessControlAction.ActionUserRolesList)) {
-        const userRoles = await fetchUserRoles(userId, orgId);
-        setAppliedRoles(userRoles);
-      } else {
-        setAppliedRoles([]);
+        return await fetchUserRoles(userId, orgId);
       }
     } catch (e) {
       // TODO handle error
       console.error('Error loading options');
     }
+    return [];
   }, [orgId, userId]);
 
+  useEffect(() => {
+    getUserRoles();
+  }, [orgId, userId, getUserRoles]);
```

Comment:
```suggestion
  useEffect(getUserRoles, [orgId, userId, getUserRoles]);
```
Since we're just calling a function, there's no need to create an extra function wrapper.

Reply:
But useEffect expects this type:
```ts
// NOTE: callbacks are _only_ allowed to return either void, or a destructor.
type EffectCallback = () => (void | Destructor);
```

Reply:
Ah ok, makes sense.